internal/renderers/skia/cached_image.rs

```diff
@@ -77,6 +80,29 @@ pub(crate) fn as_skia_image(
         ImageInner::BackendStorage(x) => {
             vtable::VRc::borrow(x).downcast::<SkiaCachedImage>().map(|x| x.image.clone())
         }
+        #[cfg(skia_backend_opengl)]
+        ImageInner::BorrowedOpenGLTexture(BorrowedOpenGLTexture { texture_id, size, .. }) => unsafe {
+            let mut texture_info = skia_safe::gpu::gl::TextureInfo::from_target_and_id(
+                glow::TEXTURE_2D,
+                texture_id.get(),
+            );
+            texture_info.format = glow::RGBA8;
+            let backend_texture = skia_safe::gpu::BackendTexture::new_gl(
+                (size.width as _, size.height as _),
+                skia_safe::gpu::Mipmapped::No,
+                texture_info,
+            );
+            skia_safe::image::Image::from_texture(
+                _canvas.recording_context().as_mut().unwrap(),
+                &backend_texture,
+                skia_safe::gpu::SurfaceOrigin::TopLeft,
+                skia_safe::ColorType::RGBA8888,
+                skia_safe::AlphaType::Unpremul,
+                None,
+            )
+        },
+        #[cfg(not(skia_backend_opengl))]
+        ImageInner::BorrowedOpenGLTexture(..) => None,
```

Comment:
In general, I wonder of we should have better error reporting.
Should we print something to the stderr?
Should we have a error callback? I think we should.
It doesn't have to be in this patch though.

Reply:
I don't think this is the right place to handle errors or notify the application. If you create a `slint::Image` from an external OpenGL texture and use it with a renderer that doesn't support it, then right place to handle this is _before_ entering this code path. If we reach this code path, I think we should not render anything. But the right place to handle this is much earlier:

1. At compile or run-time, I want to make sure that a renderer is selected that supports this API.
2. At run-time I want to be able to check early on (in main()) what the capabilities of the graphics APIs in use are, and if the APIs the application needs are not supported, then do application specific error handling.

Trying to "handle" this at the time of rendering is futile IMO.


So two examples:

1. Suppose you're writing an application that shows maps and you know that the renderer requires OpenGL. I want to make sure that compile-time or very early at run-time that Slint supports the necessary features. If it's a run-time selection, it may be necessary to abort and inform the user. (Certainly not at rendering time).

2. Suppose you're writing a video player that should work everywhere. At start-up I need to check what APIs  are available: If OpenGL is available and I can efficiently create textures, then I may want to enable that code path in the renderer. Otherwise a fallback to creating slint::Image objects from pixel buffers may make more sense.


So in conclusion I think the correct way is to offer APIs to introspect graphics capabilities and optionally influence selection.